source/Locales/Compactness/Properties.lagda

```diff
@@ -0,0 +1,716 @@
+Kuratowski-finite subcover.
+
+\begin{code}
+
+is-compact-open'' : (X : Locale 𝓤 𝓥 𝓦) → ⟨ 𝒪 X ⟩ → Ω (𝓤 ⊔ 𝓦 ⁺)
+is-compact-open'' {𝓤} {𝓥} {𝓦} X U =
+ Ɐ S ꞉ Fam 𝓦 ⟨ 𝒪 X ⟩ ,
+  (U ＝ₚ ⋁[ 𝒪 X ] S) ⇒
+   (Ǝ (J , h) ꞉ SubFam S , is-Kuratowski-finite J
+                         × (U ＝ ⋁[ 𝒪 X ] ⁅  S [ h j ] ∣ j ∶ J ⁆))
+    where
+     open PosetNotation (poset-of (𝒪 X))
+     open Equality carrier-of-[ poset-of (𝒪 X) ]-is-set
+
+module Characterization-Of-Compactness₃ (X : Locale 𝓤 𝓥 𝓦) where
+
+ open PosetNotation (poset-of (𝒪 X))
+ open PosetReasoning (poset-of (𝒪 X))
+ open Some-Lemmas-On-Directification (𝒪 X)
+
+\end{code}
+
+To see that `is-compact-open'` implies `is-compact-open''`, notice first that
+for every open `U : ⟨ 𝒪 X ⟩` and family `S`, we have that `U ≤ ⋁ S` if and
+only if `U ＝ ⋁ { U ∧ Sᵢ ∣ i : index S }`.
+
+\begin{code}
+
+ distribute-inside-cover₁ : (U : ⟨ 𝒪 X ⟩) (S : Fam 𝓦 ⟨ 𝒪 X ⟩)
+                          → U ＝ ⋁[ 𝒪 X ] ⁅ U ∧[ 𝒪 X ] (S [ i ]) ∣ i ∶ index S ⁆
+                          → (U ≤ (⋁[ 𝒪 X ] S)) holds
+ distribute-inside-cover₁ U S p = connecting-lemma₂ (𝒪 X) †
+  where
+   Ⅰ = p
+
+   Ⅱ : ⋁[ 𝒪 X ] ⁅ U ∧[ 𝒪 X ] S [ i ] ∣ i ∶ index S ⁆ ＝ U ∧[ 𝒪 X ] (⋁[ 𝒪 X ] S)
+   Ⅱ = distributivity (𝒪 X) U S ⁻¹
+
+   † : U ＝ U ∧[ 𝒪 X ] (⋁[ 𝒪 X ] S)
+   † = U                                               ＝⟨ Ⅰ ⟩
+       ⋁[ 𝒪 X ] ⁅ U ∧[ 𝒪 X ] S [ i ] ∣ i ∶ index S ⁆   ＝⟨ Ⅱ ⟩
+       U ∧[ 𝒪 X ] (⋁[ 𝒪 X ] S)                         ∎
+
+ distribute-inside-cover₂
+  : (U : ⟨ 𝒪 X ⟩) (S : Fam 𝓦 ⟨ 𝒪 X ⟩)
+  → (U ≤ (⋁[ 𝒪 X ] S)) holds
+  → U ＝ ⋁[ 𝒪 X ] ⁅ U ∧[ 𝒪 X ] (S [ i ]) ∣ i ∶ index S ⁆
+ distribute-inside-cover₂ U S p =
+  U                                                 ＝⟨ Ⅰ ⟩
+  U ∧[ 𝒪 X ] (⋁[ 𝒪 X ] S)                           ＝⟨ Ⅱ ⟩
+  ⋁[ 𝒪 X ] ⁅ U ∧[ 𝒪 X ] (S [ i ]) ∣ i ∶ index S ⁆   ∎
+  where
+   Ⅰ = connecting-lemma₁ (𝒪 X) p
+   Ⅱ = distributivity (𝒪 X) U S
+
+\end{code}
+
+The backward implication follows easily from these two lemmas.
+
+\begin{code}
+
+ compact-open''-implies-compact-open' : (U : ⟨ 𝒪 X ⟩)
+                                      → is-compact-open'' X U holds
+                                      → is-compact-open'  X U holds
+ compact-open''-implies-compact-open' U κ S p = ∥∥-functor † ♢
+  where
+   q : U ＝ ⋁[ 𝒪 X ] ⁅ U ∧[ 𝒪 X ] (S [ i ]) ∣ i ∶ index S ⁆
+   q = distribute-inside-cover₂ U S p
+
+   ♢ : ∃ (J , h) ꞉ SubFam S , is-Kuratowski-finite J
+                            × (U ＝ ⋁[ 𝒪 X ] ⁅ U ∧[ 𝒪 X ] S [ h j ] ∣ j ∶ J ⁆)
+   ♢ = κ ⁅ U ∧[ 𝒪 X ] (S [ i ]) ∣ i ∶ index S ⁆ q
+
+   † : Σ (J , h) ꞉ SubFam S , is-Kuratowski-finite J
+                            × (U ＝ ⋁[ 𝒪 X ] ⁅ U ∧[ 𝒪 X ] S [ h j ] ∣ j ∶ J ⁆)
+     → Σ (J , h) ꞉ SubFam S , is-Kuratowski-finite J
+                            × (U ≤ (⋁[ 𝒪 X ] ⁅ S [ h j ] ∣ j ∶ J ⁆)) holds
+   † (𝒥@(J , h) , 𝕗 , p) =
+    𝒥 , 𝕗 , distribute-inside-cover₁ U ⁅ S [ h j ] ∣ j ∶ J ⁆ p
+
+\end{code}
+
+Now, the forward implication, which is a bit more involved.
```

Comment:
I think this is much simpler to prove actually: assume you have $U = \bigvee S$. By primed compactness, you get a Kuratowski finite $F \subseteq S$ with $U \le {\bigvee F}$. It remains to show that $U = {\bigvee F}$, but $\bigvee F \le \bigvee S = U$.